protocol/bc/value.go

```diff
@@ -13,6 +15,53 @@ type ValueSource struct {
 	Entry `entry:"-"`
 }
 
+func (vs *ValueSource) CheckValid(vstate *validationState) error {
```

Comment:
Ha! No. I originally had all the `CheckValid` functions exported, and at some point I unexported them with:

`gorename -from '"chain/protocol/bc".ValidChecker.CheckValid' -to checkValid`

(I then also unexported `validChecker`.)

As I now understand, this worked on `Entry` types like `Mux` and `Spend` because the definition of interface `Entry` explicitly includes `validChecker`, but it skipped `ValueSource` and `ValueDestination` because nothing says that those types have to be `validChecker`s. Interesting!